jax/api_util.py

```diff
@@ -124,6 +142,49 @@ def _argnums_partial(dyn_argnums, fixed_args, *dyn_args, **kwargs):
   ans = yield args, kwargs
   yield ans
 
+
+def argnames_partial(f, dyn_argnames, kwargs):
+  dyn_argnames = _ensure_str_tuple(dyn_argnames)
+  fixed_kwargs = tuple((k, unit if k in dyn_argnames else wrap_hashably(v))
+                       for k, v in kwargs.items())
+  dyn_kwargs = {k: kwargs[k] for k in dyn_argnames}
+  return _argnames_partial(f, WrapKwArgs(fixed_kwargs)), dyn_kwargs
+
+
+def argnames_partial_except(f: lu.WrappedFun, static_argnames: Tuple[str, ...],
+                            kwargs: Dict[str, Any]):
+  if not static_argnames:
+    return f, kwargs
+  dyn_argnames = tuple(k for k in kwargs if k not in static_argnames)
+  dyn_kwargs = {k: kwargs[k] for k in dyn_argnames}
```

Comment:
Combine these comprehensions and iterate over `kwargs.items()`?

Reply:
Done!